src/mlpack/methods/ann/layer/padding_impl.hpp

```diff
@@ -42,10 +46,29 @@ void Padding<InputDataType, OutputDataType>::Forward(
 {
   nRows = input.n_rows;
   nCols = input.n_cols;
-  output = arma::zeros(nRows + padWLeft + padWRight,
-      nCols + padHTop + padHBottom);
-  output.submat(padWLeft, padHTop, padWLeft + nRows - 1,
-      padHTop + nCols - 1) = input;
+  
+  if (inputWidth == 0 || inputHeight == 0)
+  {
+    output = arma::zeros(nRows + padWLeft + padWRight,
+        nCols + padHTop + padHBottom);
+    output.submat(padWLeft, padHTop, padWLeft + nRows - 1,
+        padHTop + nCols - 1) = input;
+  }
+  else
+  {
+    inSize = input.n_elem / (inputWidth * inputHeight * nCols);
+    inputTemp = arma::cube(const_cast<arma::Mat<eT>&>(input).memptr(),
+        inputWidth, inputHeight, nCols * inSize, false, false);
+    outputTemp = arma::zeros<arma::Cube<eT> >(inputWidth + padWLeft + padWRight,
+        inputHeight + padHTop + padHBottom, nCols * inSize);
+    outputTemp.subcube(padWLeft, padHTop, 0, padWLeft + inputWidth - 1,
+        padHTop + inputHeight - 1, inSize - 1) = inputTemp;
```

Comment:
Let me know if this makes sense, number of slices above nCols * inSize, but in the subcube the last slice is inSize - 1. The code should be you iterate over all columns i.e. batches and shape them as cube and the pad them.

Reply:
that does make sesne, so for now if i assume that there was only a single batch which i have in the test case it should work, right ?

Reply:
Lets first fix the implementation and the we can see if there is an issue with the cube conversion.

Reply:
Sure, Best of Luck!

Reply:
Thanks :hugs: